app/views/trips/_trip.html.haml

```diff
@@ -1,18 +1,21 @@
 .trip
-  = link_to trip, class: 'trip-item' do
+  = link_to trip, class: [('trip-item-before' if (trip.is_before_today? || trip.is_strictly_before(@search.date_value))), ('trip-item-after' if trip.is_strictly_after(@search.date_value)), ('trip-item')] do
     .driver
       .avatar= image_tag('covoiturage-avatar-default2.png')
       .info
         .name= trip.name
         .age= "#{trip.age} ans" if trip.age.present?
     .trip-info
       .title
-        =l trip.departure_date, format: :long
+        =l trip.departure_date, format: :trip_date
         =l trip.departure_time, format: :short
+        %p= "(le covoit' est déjà parti !)" if (trip.is_before_today?) 
+        %p= "/!\\ départ dans #{Integer(trip.departure_date - @search.date_value)} jour(s)" if (trip.is_strictly_before(@search.date_value)) 
```

Comment:
Does this outputs negative time intervals ? In that case I see the better option:Départ x jours avant votre recherche.

Reply:
For the negative, there is the `is_strictly_before ` and search filters all trips before selected date.
But it could be a good idea if no result returned to display them with a message liek that.